### PR TITLE
Restage cli

### PR DESCRIPTION
**Pull request by enrichman**
This PR adds the `restage` command to the cli.

While trying to add some tests I've encountered some difficulties, so I'm going to open a couple of smaller PRs to make it easier to understand.

Adding ApiClient interface:
 * https://github.com/epinio/epinio/pull/1289

Moving AppLogs from EpinioClient to the ApiClient
* https://github.com/epinio/epinio/pull/1291

### CODE REVIEW

File: internal/cli/apps.go [hunk 1]
```diff
@@ -274,8 +275,9 @@ var CmdAppManifest = &cobra.Command{
 
 // CmdAppRestart implements the command: epinio app restart
 var CmdAppRestart = &cobra.Command{
-	Use:               "restart NAME",
-	Short:             "Restart the application",
+	Use:   "restart NAME",
+	Short: "Restart the application",
+
```
Comment (andreas-kupries) [inline internal/cli/apps.go:280]: Superfluous empty line, disrupts vertical alignment.

File: internal/cli/usercmd/app.go [hunk 2]
```diff
@@ -526,3 +526,45 @@ func (c *EpinioClient) printReplicaDetails(app models.App) error {
...
+
+	log.V(1).Info("restaging application")
+
+	app, err := c.API.AppShow(c.Settings.Namespace, appName)
+	if err != nil {
+		return err
+	}
+
+	if app.Origin.Kind == models.OriginContainer {
+		return nil
```
Comment (andreas-kupries) [inline internal/cli/usercmd/app.go:553]: IMHO this should print a note, i.e. `c.ui.Note().Msg("Unable to restage container-based application")`, or similar.

File: internal/cli/usercmd/app.go [hunk 3]
```diff
@@ -526,3 +526,45 @@ func (c *EpinioClient) printReplicaDetails(app models.App) error {
...
+	}
+
+	c.ui.Normal().Msg("Restaging application...")
```
Comment (andreas-kupries) [inline internal/cli/usercmd/app.go:556]: I see no need to essentially repeat the message from lines 536-539 above.

File: internal/cli/usercmd/app_test.go [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+			It("returns no error", func() {
+				epinioClient, err := usercmd.NewEpinioClient(&settings.Settings{Namespace: "workspace"}, mockClient)
+				Expect(err).ToNot(HaveOccurred())
+
+				err = epinioClient.AppRestage("appname")
+				Expect(err).ToNot(HaveOccurred())
+			})
+		})
+
+		When("restaging an app of kind container", func() {
```
Comment (andreas-kupries) [inline internal/cli/usercmd/app_test.go:51]:
```suggestion
		When("restaging a container-based app", func() {
```

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
Just a number of nits.

File: internal/cli/apps.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/cli/apps.go]: Yeah, go-format auto-formatting

**Review by andreas-kupries — DISMISSED** [review]

**Comment by andreas-kupries** [comment]
Ah! While the new code has unit tests, there are no new CI cli tests, right ? That should be fixed, IMHO.
As it uses the existing stage endpoint new api tests are __not__ required.

File: acceptance/apps_test.go [hunk 5]
```diff
@@ -215,6 +215,57 @@ configuration:
...
+					appName,
+					"--name", appName,
+					"--git", wordpress+",main")
+				Expect(err).ToNot(HaveOccurred(), pushLog)
+
+				Eventually(func() string {
+					out, err := env.Epinio("", "app", "list")
+					Expect(err).ToNot(HaveOccurred(), out)
+					return out
+				}, "5m").Should(MatchRegexp(fmt.Sprintf(`%s.*\|.*1\/1.*\|.*`, appName)))
```
Comment (andreas-kupries) [inline acceptance/apps_test.go:232]: We might have a helper function creating an app and waiting for it to be available.

File: acceptance/apps_test.go [hunk 6]
```diff
@@ -215,6 +215,57 @@ configuration:
...
+					appName,
+					"--name", appName,
+					"--container-image-url", containerImageURL)
+				Expect(err).ToNot(HaveOccurred(), pushLog)
+
+				Eventually(func() string {
+					out, err := env.Epinio("", "app", "list")
+					Expect(err).ToNot(HaveOccurred(), out)
+					return out
+				}, "5m").Should(MatchRegexp(fmt.Sprintf(`%s.*\|.*1\/1.*\|.*`, appName)))
```
Comment (andreas-kupries) [inline acceptance/apps_test.go:255]: We should have a helper function creating a container-based app and waiting for it to be available.

acceptance/helpers/machine/apps.go:20:func (m *Machine) MakeContainerImageApp(appName string, instances int, containerImageURL string) string {

**Review by andreas-kupries — CHANGES_REQUESTED** [review]
App creation could be simplified.

File: acceptance/apps_test.go (hunk 6 above)
Comment (andreas-kupries) [reply acceptance/apps_test.go:255]: And
> acceptance/apps_test.go:593:			env.MakeContainerImageApp(appName, 1, containerImageURL)
for example use

File: acceptance/apps_test.go (hunk 5 above)
Comment (andreas-kupries) [reply acceptance/apps_test.go:232]: See acceptance/helpers/machine/apps.go

Comment (enrichman) [reply acceptance/apps_test.go:232]: Ah, much easier! Thanks

File: acceptance/apps_test.go (hunk 6 above)
Comment (enrichman) [reply acceptance/apps_test.go:255]: Fixing! Much better

**Review by andreas-kupries — APPROVED** [review]

**Pull request merged**